docs/references/config/config.toml.md

```diff
@@ -0,0 +1,913 @@
+
+### p2p.max_num_inbound_peers
+Maximum number of inbound peers.
+```toml
+max_num_inbound_peers = 40
+```
+
+| Value type          | integer |
+|:--------------------|:--------|
+| **Possible values** | &gt;= 0 |
+
+The [`p2p.max_num_inbound_peers`](#p2pmax_num_inbound_peers) and
+[`p2p.max_num_outbound_peers`](#p2pmax_num_outbound_peers) values together define how many P2P connections the node will
+maintain at maximum capacity, not including list of nodes in the [`p2p.persistent_peers`](#p2ppersistent_peers) setting.
+
+The connections are bidirectional, so any connection can send or receive blocks and other data. The separation into
+inbound and outbound setting only distinguishes the initial setup of the connection: outbound connections are initiated
+by the node while inbound connections are initiated by a remote party.
+
+Nodes on non-routable networks have to set their gateway to port-forward the P2P port for inbound connections to reach
+the node. Outbound connections can be initiated as long as the node has generic Internet access. (Using NAT or methods.)
+
+### p2p.max_num_outbound_peers
+Maximum number of outbound peers.
+```toml
+max_num_outbound_peers = 10
+```
+
+| Value type          | integer |
+|:--------------------|:--------|
+| **Possible values** | &gt;= 0 |
+
+The [`p2p.max_num_inbound_peers`](#p2pmax_num_inbound_peers) and
+[`p2p.max_num_outbound_peers`](#p2pmax_num_outbound_peers) values together define how many P2P connections the node will
+maintain at maximum capacity, not including list of nodes in the [`p2p.persistent_peers`](#p2ppersistent_peers) setting.
+
+The connections are bidirectional, so any connection can send or receive blocks and other data. The separation into
+inbound and outbound setting only distinguishes the initial setup of the connection: outbound connections are initiated
+by the node while inbound connections are initiated by a remote party.
+
+Nodes on non-routable networks have to set their gateway to port-forward the P2P port for inbound connections to reach
+the node. Outbound connections can be initiated as long as the node has generic Internet access. (Using NAT or methods.)
+
+### p2p.unconditional_peer_ids
+List of node IDs, to which a connection will be (re)established ignoring any existing limits.
+```toml
+unconditional_peer_ids = ""
+```
+
+| Value type          | string (comma-separated)         |
+|:--------------------|:---------------------------------|
+| **Possible values** | comma-separated list of node IDs |
+|                     | `""`                             |
+
+If a peer listed in this property requests a connection, it will be accepted, even if the
+[`p2p.max_num_inbound_peers`](#p2pmax_num_inbound_peers) or the
+[`p2p.max_num_outbound_peers`](#p2pmax_num_outbound_peers) are reached.
+
+Contrary to other settings, only the node ID has to be defined here, not the IP:port of the remote node.
+
+### p2p.persistent_peers_max_dial_period
```

Comment:
I don't think our p2p layer uses this. The intervals used here are constants:

https://github.com/cometbft/cometbft/blob/main/p2p/switch.go#L27

Reply:
I thought it's used here: https://github.com/cometbft/cometbft/blob/f128c623cb58a6813cd0742565351485c0fa5dbb/node/setup.go#L502C4-L502C32

Then it is used in `maxBackoffDurationForPeer` at https://github.com/cometbft/cometbft/blob/f128c623cb58a6813cd0742565351485c0fa5dbb/p2p/pex/pex_reactor.go#L598

Which is used in `dialAttemptsInfo` to determine the backoff time.

Reply:
I see. This is used when dialing it at first. The `Switch` method when reconnecting. Why are them distinct? No idea.